src/hio.rs

```diff
@@ -68,12 +68,16 @@ fn write_all(fd: usize, mut buffer: &[u8]) -> Result<(), ()> {
             // Done
             0 => return Ok(()),
             // `n` bytes were not written
-            n if n <= buffer.len() => {
+            n if n <= buffer.len() && n > 0 => {
```

Comment:
Since `n: usize` and we've already matched `n==0`, what's the purpose of the `n>0` check?

Reply:
Indeed, copy-pasta from his patch. Removing.

Reply:
This was in order to cast the u32 to i32 and then reject all negative numbers, but was never finished